src/meta/processors/admin/BalanceProcessor.cpp

```diff
@@ -85,21 +88,21 @@ void BalanceProcessor::process(const cpp2::BalanceReq& req) {
         onFinished();
         return;
     }
-    std::unordered_set<HostAddr> hostDel;
-    if (req.get_host_del() != nullptr) {
-        hostDel.reserve(req.get_host_del()->size());
-        for (const auto& host : *req.get_host_del()) {
-            hostDel.emplace(host);
-        }
-    }
+
     auto hosts = ActiveHostsMan::getActiveHosts(kvstore_);
     if (hosts.empty()) {
         LOG(ERROR) << "There is no active hosts";
         handleErrorCode(cpp2::ErrorCode::E_NO_HOSTS);
         onFinished();
         return;
     }
-    auto ret = Balancer::instance(kvstore_)->balance(std::move(hostDel));
+
+    std::vector<HostAddr> lostHosts;
+    if (req.__isset.host_del) {
+        lostHosts = *req.get_host_del();
+    }
```

Comment:
I have a question，
When `lostHosts `is empty,   here should return directly？

Reply:
That's not the truth. when the lost hosts is empty we will try to balance another hosts ~

Reply:
Got，thx